cmd/zconfigd/zconfigd.c

```diff
@@ -0,0 +1,95 @@
+#include <unistd.h>
+#include <CoreFoundation/CoreFoundation.h>
+#include <IOKit/IOKitLib.h>
+#include <mach/mach.h>
+
+#define kNetLundmanZfsZvol		"net_lundman_zfs_zvol"
+
+// globals
+static IONotificationPortRef    gNotifyPort;
+static io_iterator_t            gKextLoadedIter;
+
+void SignalHandler(int sigraised)
+{
+	printf("\nInterrupted\n");
+
+	// Clean up here
+	IONotificationPortDestroy(gNotifyPort);
+
+	if (gKextLoadedIter)
+	{  
+		IOObjectRelease(gKextLoadedIter);
+		gKextLoadedIter = 0;
+	}
+
+	// exit(0) should not be called from a signal handler.  Use _exit(0) instead
+	//
+	_exit(0);
```

Comment:
The tradition is to set a "do_exit" variable high in the interrupt handler and not call _exit(). But you can't as you are in a RunLoop, I believe you are supposed to call CFRunLoopStop() in that case. But if that doesn't work, _exit() will do.